.github/workflows/build.yml

```diff
@@ -98,3 +98,38 @@ jobs:
       - name: Setup tmate debug session
         uses: mxschmitt/action-tmate@v3
         if: env.RUN_TMATE
+  test:
+    name: test source
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Download Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '^1.20.2'
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: Retrieve package location
+        run: |
+            echo PACKAGE_LOCATION="$(find ./ -name 'test')/../" >> $GITHUB_OUTPUT
```

Comment:
@mcdonnnj You know this stuff better than the rest of us.  Is this how we want to be setting the `PACKAGE_LOCATION` or can you think of a better way to do it?

Reply:
Go packages in general have their go files within the root of a project/repository. Considering this is a skeleton and example I opted to place everything in `src`. When this is built it's looking in the root of the repository for the go files. So to not hard-code the path I opted to set a variable to a location outside of the test directory which is the root of our package.

Reply:
If Go packages have files in the root as a standard then I think we should follow suite. Thoughts @dav3r @jsf9k ?

Reply:
So @jasonodoom put the source files in `src/` which necessitates this bit of hackery. If a Golang project has the Go source files in the root of the repository as a standard thing then I think we should follow suite.

Reply:
Agreed

Reply:
test

Reply:
It's not enforced and both ways [work](https://go.dev/doc/gopath_code) But in the case of a Go module such as [unistyle](https://github.com/danielgtaylor/unistyle) the src directory isn't used. It's personal preference and I've seen both ways. Mainly that directory structure for projects 

@mcdonnnj I want to clarify that this is also an example and also serves to present the directory structure for a project that is not a Go module.

Reply:
My understanding is that you should prefer to develop as a Go module unless you explicitly have a reason not to do so these days. It seems like the project layout described in [golang-standards/project-layout](https://github.com/golang-standards/project-layout) is pretty popular (but unofficial!) as it's used by [kubernetes](https://github.com/kubernetes/kubernetes), [Docker](https://github.com/moby/moby), and other high visibility Go projects. Major projects that do not follow the pure module layout tend to have some core Go source files in the root of the directory such as in [Gitea](https://github.com/go-gitea/gitea), [Rclone](https://github.com/rclone/rclone), and [Terraform](https://github.com/hashicorp/terraform).

Since there appear to be no firm Go packaging guidelines (unlike with like [Python](https://packaging.python.org/en/latest/)) I would like us to mirror mainstream Go development best practices/patterns.

Reply:
gotcha. I will make those changes then 😭

Reply:
Please see commit https://github.com/cisagov/skeleton-golang-package/pull/2/commits/5428611a9c522e1102c4aade9288ccfb67990ae1

Reply:
LGTM along with 6ef5cc3eb584a30042cef6961279e2b891195a1c to update the build workflow.